examples/react/optimistic-updates-typescript/pages/index.tsx

```diff
@@ -35,41 +35,8 @@ function Example() {
   const { isFetching, ...queryInfo } = useTodos()
 
   const addTodoMutation = useMutation({
-    mutationFn: (newTodo) => axios.post('/api/data', { text: newTodo }),
-    // When mutate is called:
-    onMutate: async (newTodo: string) => {
-      setText('')
-      // Cancel any outgoing refetches
-      // (so they don't overwrite our optimistic update)
-      await queryClient.cancelQueries({ queryKey: ['todos'] })
-
-      // Snapshot the previous value
-      const previousTodos = queryClient.getQueryData<Todos>(['todos'])
-
-      // Optimistically update to the new value
-      if (previousTodos) {
-        queryClient.setQueryData<Todos>(['todos'], {
-          ...previousTodos,
-          items: [
-            ...previousTodos.items,
-            { id: Math.random().toString(), text: newTodo },
-          ],
-        })
-      }
-
-      return { previousTodos }
-    },
-    // If the mutation fails,
-    // use the context returned from onMutate to roll back
-    onError: (err, variables, context) => {
-      if (context?.previousTodos) {
-        queryClient.setQueryData<Todos>(['todos'], context.previousTodos)
-      }
-    },
-    // Always refetch after error or success:
-    onSettled: () => {
-      queryClient.invalidateQueries({ queryKey: ['todos'] })
-    },
+    mutationFn: (newTodo: string) => axios.post('/api/data', { text: newTodo }),
+    onSettled: () => queryClient.invalidateQueries({ queryKey: ['todos'] }),
```

Comment:
We could potentially add to this example a comment that `mutation` variables would be available in your component until `invalidation` Promise is finished

Reply:
yeah I'd definitely need to re-write docs to this approach